pkg/apis/serving/v1/revision_defaults.go

```diff
@@ -158,6 +162,37 @@ func (*RevisionSpec) applyProbes(container *corev1.Container) {
 	}
 }
 
+//
+func (rs *RevisionSpec) defaultSecurityContext(ctx context.Context, container *corev1.Container, cfg *config.Config) {
+	if cfg.Features.SecurePodDefaults != config.Enabled {
+		return
+	}
+
+	if container.SecurityContext == nil {
+		container.SecurityContext = &corev1.SecurityContext{}
+	}
+
+	if container.SecurityContext.AllowPrivilegeEscalation == nil {
+		container.SecurityContext.AllowPrivilegeEscalation = ptr.Bool(false)
+	}
+	if container.SecurityContext.SeccompProfile == nil {
+		container.SecurityContext.SeccompProfile = &corev1.SeccompProfile{}
+	}
+	if container.SecurityContext.SeccompProfile.Type == "" {
+		container.SecurityContext.SeccompProfile.Type = corev1.SeccompProfileTypeRuntimeDefault
+	}
```

Comment:
We don't allow `SeccompProfile` thru the fieldmask today (it looks like it may be newer than our field triage because it isn't copied or explicitly cleared): https://github.com/knative/serving/blob/a18077cd9bbfe9e6208b2202fc39dfa3c79fc837/pkg/apis/serving/fieldmask.go#L623-L639

So my main concern here would be that we are changing a default value that we don't allow folks to override.

Reply:
Good point, I can make a PR to make this settable.

Reply:
#13401